### PR TITLE
Ak/user event notification

### PR DESCRIPTION
**Pull request by mitochondrion**

### CODE REVIEW

File: ExampleApp/AppDelegate.swift [hunk 1]
```diff
@@ -67,6 +69,64 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+            name: TradeItNotification.Name.errorShown,
+            object: nil
+        )
+    }
+
+    func onErrorShownNotification(notification: Notification) {
+        let view = notification.userInfo?[TradeItNotification.UserInfoKey.view] ?? "NO KEY FOR VIEW"
+        let errorTitle = notification.userInfo?[TradeItNotification.UserInfoKey.errorTitle] ?? "NO KEY FOR ERROR CODE"
+        let errorMessage = notification.userInfo?[TradeItNotification.UserInfoKey.errorMessage] ?? "NO KEY FOR ERROR MESSAGE"
+        print("=====> ERROR SHOWN: VIEW: \(view), TITLE: \(errorTitle), MESSAGE: \(errorMessage)")
```
Comment (jsom) [inline ExampleApp/AppDelegate.swift]: Do we want to just pass the error object back here?

File: ExampleApp/Base.lproj/Main.storyboard [hunk 2]
```diff
@@ -72,5 +72,53 @@
             </objects>
             <point key="canvasLocation" x="-678" y="-290"/>
         </scene>
+        <!--Example View Controller-->
+        <scene sceneID="uFO-j0-SXY">
```
Comment (jsom) [inline ExampleApp/Base.lproj/Main.storyboard]: Why is this VC new?

File: TradeItIosTicketSDK2/Constants.swift [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    public static let nameViewDidAppear = TradeItNotification.Name.viewDidAppear
+    public static let nameErrorShown = TradeItNotification.Name.errorShown
+
+    // MARK: Notification UserInfo keys
+    public static let userInfoKeyView = TradeItNotification.UserInfoKey.view
+    public static let userInfoKeyErrorTitle = TradeItNotification.UserInfoKey.errorTitle
+    public static let userInfoKeyErrorMessage = TradeItNotification.UserInfoKey.errorMessage
+}
+
+public struct TradeItNotification {
```
Comment (jsom) [inline TradeItIosTicketSDK2/Constants.swift]: I don't think Swift structs are available in obj-c

File: TradeItIosTicketSDK2/TradeItYahooViewController.swift [hunk 4]
```diff
@@ -19,4 +19,16 @@ class TradeItYahooViewController: CloseableViewController {
             containerView.bottomAnchor.constraint(equalTo: self.view.bottomAnchor)
         ])
     }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        NotificationCenter.default.post(
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItYahooViewController.swift]: Nice

File: ExampleApp/AppDelegate.swift (hunk 1 above)
Comment (mitochondrion) [reply ExampleApp/AppDelegate.swift]: That makes sense.

File: ExampleApp/Base.lproj/Main.storyboard (hunk 2 above)
Comment (mitochondrion) [reply ExampleApp/Base.lproj/Main.storyboard]: To fix some silliness that occurs when reusing the storyboard nib for advanced options screen.

File: TradeItIosTicketSDK2/Constants.swift (hunk 3 above)
Comment (mitochondrion) [reply TradeItIosTicketSDK2/Constants.swift]: See the @objc class above.

**Pull request merged**